File: ios/lib/ReactNativeCameraKit/CKOverlayObject.m
#import "CKOverlayObject.h"

@interface CKOverlayObject ()

@property (nonatomic, readwrite) float width;
@property (nonatomic, readwrite) float height;
@property (nonatomic, readwrite) float ratio;

@end

@implementation CKOverlayObject

-(instancetype)initWithString:(NSString*)str {

    self = [super init];

    if (self) {
        [self commonInit:str];
    }

    return self;
}

-(void)commonInit:(NSString*)str {

    NSArray<NSString*> *array = [str componentsSeparatedByString:@":"];
    if (array.count == 2) {
        float height = [array[0] floatValue];
        float width = [array[1] floatValue];

        if (width != 0 && height != 0) {
            self.width = width;
            self.height = height;
            self.ratio = self.width/self.height;
        }
    }
}

-(NSString *)description {
    return [NSString stringWithFormat:@"width:%f height:%f ratio:%f", self.width, self.height, self.ratio];
}


@end

File: ios/ReactNativeCameraKit/CKOverlayObject.swift
//
//  CKOverlayObject.swift
//  ReactNativeCameraKit
//

import Foundation

struct CKOverlayObject: CustomStringConvertible {
    let width: Float
    let height: Float
    let ratio: Float

    init(from inputString: String) {
        let values = inputString.split(separator: ":")

        if values.count == 2,
           let inputHeight = Float(values[0]),
           let inputWidth = Float(values[1]),
           inputHeight != 0,
           inputWidth != 0 {
                height = inputHeight
                width = inputWidth
                ratio = width / height
        } else {
            height = 0
            width = 0
            ratio = 0
        }
    }

    // MARK: CustomStringConvertible

    var description: String {
        return "height:\(height) width:\(width) ratio:\(ratio)"
    }
}
